blobs.go

```diff
@@ -146,6 +146,10 @@ type BlobIngester interface {
 
 	// Resume attempts to resume a write to a blob, identified by an id.
 	Resume(ctx context.Context, id string) (BlobWriter, error)
+
+	// Mount adds a blob to this service from another source repository,
+	// identified by a digest.
+	Mount(ctx context.Context, sourceRepo Repository, dgst digest.Digest) (Descriptor, error)
```

Comment:
@BrianBland Word.

@aaronlehmann Has refactoring PR that we may want to get in first. Up to you.